packages/keymaps/src/browser/keybindings-widget.tsx

```diff
@@ -0,0 +1,202 @@
+    }
+
+    protected searchFilter(item: KeybindingItem, query: string): boolean {
+        if (item.command.toLocaleLowerCase().includes(query)) {
+            return true;
+        } else if ((item.keybinding) && item.keybinding.toLocaleLowerCase().includes(query)) {
+            return true;
+        } else if ((item.context) && item.context.toLocaleLowerCase().includes(query)) {
+            return true;
+        } else if (item.id.toLocaleLowerCase().includes(query)) {
+            return true;
+        }
+        return false;
+    }
+
+    protected render(): React.ReactNode {
+        let content: React.ReactNode;
+        content = <React.Fragment>
+            {this.renderSearch()}
+            {(this.items.length > 0) ? this.renderTable() : this.renderMessage()}
+        </React.Fragment>;
+        return content;
+    }
+
+    protected renderSearch(): React.ReactNode {
+        return <React.Fragment>
+            <div className='search-keybindings-container'>
+                <input id='search-keybindings'
+                    className={(this.items.length > 0) ? '' : 'no-keybindings'}
+                    type='text' placeholder='Search keybindings' onKeyUp={e => this.searchKeybindings()}></input >
+            </div>
+        </React.Fragment>;
+    }
+
+    protected renderMessage(): React.ReactNode {
+        return <React.Fragment>
+            <div className='search-notification'>
+                <div>No results found!</div>
+            </div>
+        </React.Fragment>;
+    }
+
+    protected renderTable(): React.ReactNode {
+        return <React.Fragment>
+            <div className='keybindings'>
+                <table>
+                    <thead>
+                        {this.renderHeader()}
+                    </thead>
+                    <tbody>
+                        {this.renderRows()}
+                    </tbody>
+                </table>
+            </div>
+        </React.Fragment>;
+    }
+
+    protected renderHeader(): React.ReactNode {
+        return <React.Fragment>
+            <tr>
+                <th>Label</th>
+                <th>Keybinding</th>
+                <th>Command</th>
+                <th>Context</th>
+            </tr>
+        </React.Fragment>;
+    }
+
+    protected renderRows(): React.ReactNode {
+        return <React.Fragment>
+            {
+                this.items.map(item =>
+                    <tr key={item.id}>
+                        <td>{item.command}</td>
+                        <td className='monaco-keybinding'>
+                            {item.keybinding !== '' ? this.renderKeybinding(item.keybinding) : ''}
+                        </td>
+                        <td><code>{item.id}</code></td>
+                        <td><code>{item.context}</code></td>
+                    </tr>
+                )
+            }
+        </React.Fragment>;
+    }
+
+    protected renderKeybinding(keybinding?: string): React.ReactNode {
```

Comment:
I am using monaco styling, just that I felt like vscode, black keybindings looked better on the dark theme to me. Perhaps I can change the global style so that all our keybindings across the platform are similiar

Reply:
I mean, for instance, Monaco displays some keys like locale-specific icons, i.e. ctrlcmd, alt, shift and so on.

Reply:
I've been thinking of doing that as well, but perhaps in an additional PR and it can be re-used across Theia

Reply:
Sounds good, maybe you can open an issue with a list of all features which should be added later?

Reply:
Sure will do!